tokio-process/src/lib.rs

```diff
@@ -362,26 +371,30 @@ impl CommandExt for Command {
         self.stdout(Stdio::piped());
         self.stderr(Stdio::piped());
 
-        let inner = self
-            .spawn_async_with_handle(handle)
-            .into_future()
+        let inner = futures::future::ready(self.spawn_async_with_handle(handle))
             .and_then(Child::wait_with_output);
 
         OutputAsync {
-            inner: Box::new(inner),
+            inner: inner.boxed(),
         }
     }
 }
 
 /// A drop guard which ensures the child process is killed on drop to maintain
 /// the contract of dropping a Future leads to "cancellation".
 #[derive(Debug)]
-struct ChildDropGuard<T: Kill> {
+struct ChildDropGuard<T: Kill>
+where
+    T: Unpin,
```

Comment:
Nit: can we combine all bounds in the where clause instead of splitting them up inline and in a where clause? (applies to rest of the declarations too)